lib/spack/spack/spec.py

```diff
@@ -188,8 +188,10 @@
                  '^': dependency_color,
                  '#': hash_color}
 
-"""Regex used for splitting by spec field separators."""
-_separators = '[%s]' % ''.join(color_formats.keys())
+"""Regex used for splitting by spec field separators.
+These need to be escaped to avoid metacharacters in color_formats.keys().
+"""
```

Comment:
@tgamblin For that we used:
```python
#: Regex used for splitting by spec field separators.
#: These need to be escaped to avoid metacharacters in color_formats.keys().
```
consistently in the `build_systems` folder (comments are placed *before* the variable they document, docstrings *after*) .